openwillis/measures/video/eye_blink.py

```diff
@@ -0,0 +1,285 @@
+    """
+    results = face_mesh.process(cv2.cvtColor(frame, cv2.COLOR_BGR2RGB))
+    if results.multi_face_landmarks:
+        face_landmarks = results.multi_face_landmarks[0]
+        leftEye = np.array([(face_landmarks.landmark[lidx].x, face_landmarks.landmark[lidx].y) for lidx in LEFT_EYE_INDICES], dtype=np.float32)
+        rightEye = np.array([(face_landmarks.landmark[ridx].x, face_landmarks.landmark[ridx].y) for ridx in RIGHT_EYE_INDICES], dtype=np.float32)
+        return leftEye, rightEye
+    return None, None
+
+
+def detect_blinks(framewise, prominence, width):
+    """
+    ---------------------------------------------------------------------------------------------------
+
+    This function detects the blinks from a given EAR array.
+
+    Parameters:
+    ............
+    framewise : pd.DataFrame
+        Contains the frame number and the eye aspect ratio (EAR) of each frame
+    prominence : float
+        The prominence of the peaks
+    width : float
+        The width of the peaks
+
+    Returns:
+    ............
+    troughs : array
+        Array containing the frame number of each blink
+    left_ips : array
+        Array containing the frame number of the start of each blink
+    right_ips : array
+        Array containing the frame number of the end of each blink
+
+    ---------------------------------------------------------------------------------------------------
+    """
+    troughs, properties = find_peaks(-framewise['EAR'], prominence=prominence, width=width)
+    left_ips = properties["left_ips"]
+    right_ips = properties["right_ips"]
+    # round to nearest integer and add 1 to match frame number
+    left_ips = np.round(left_ips).astype(int) + 1
+    right_ips = np.round(right_ips).astype(int) + 1
+    troughs = np.round(troughs).astype(int) + 1
+    return troughs, left_ips, right_ips
+
+
+def convert_frame_to_time(troughs, left_ips, right_ips, fps):
+    """
+    ---------------------------------------------------------------------------------------------------
+
+    This function converts the frame number to time.
+
+    Parameters:
+    ............
+    troughs : array
+        Array containing the frame number of each blink
+    left_ips : array
+        Array containing the frame number of the start of each blink
+    right_ips : array
+        Array containing the frame number of the end of each blink
+    fps : float
+        The fps of the video
+
+    Returns:
+    ............
+    blinks : pd.DataFrame
+        Contains for each blink the frame number and the time (in seconds)
+         start of the blink and end of the blink
+
+    ---------------------------------------------------------------------------------------------------
+    """
+    troughs_time = troughs/fps
+    left_ips_time = left_ips/fps
+    right_ips_time = right_ips/fps
+
+    blinks = pd.DataFrame({'Blink Peak Frame': troughs, 'Blink Starting Frame': left_ips, 'Blink Ending Frame': right_ips,
```

Comment:
- use config file to define variable name, in case this variables name is used at multiple places, then we just need to update config file at one place
- To be consistent with rest of the variable name, I will suggest to remove blank space from variable name instead add "_"